src/util/instanceEdit.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { getInstanceEditValues, getInstancePayload } from "./instanceEdit";
+import { LxdInstance } from "types/instance";
+
+describe("conversion to form values and back with getInstanceEditValues and getInstancePayload", () => {
+  it("preserves custom main level field", () => {
+    type CustomPayload = LxdInstance & { "custom-key": string };
+    const instance = {
+      config: {},
+      devices: {},
+      "custom-key": "custom-value",
+    } as unknown as LxdInstance;
+
+    const formValues = getInstanceEditValues(instance);
+    const payload = getInstancePayload(instance, formValues) as CustomPayload;
+
+    expect(payload["custom-key"]).toBe("custom-value");
+  });
+
+  it("preserves custom config field", () => {
```

Comment:
maybe it reads better with 'preserves custom config level instance setting field'?